aggregator/src/aggregator.rs

```diff
@@ -1430,6 +1502,35 @@ impl VdafOps {
                     )
                     .get_encoded(),
                 )
+            };
+
+            let check_keypairs = if task_hpke_keypair.is_none() && global_hpke_keypair.is_none() {
+                info!(
+                    config_id = %report_share.encrypted_input_share().config_id(),
+                    "Helper encrypted input share references unknown HPKE config ID"
+                );
+                aggregate_step_failure_counter.add(
+                    &Context::current(),
+                    1,
+                    &[KeyValue::new("type", "unknown_hpke_config_id")],
+                );
+                Err(ReportShareError::HpkeUnknownConfigId)
+            } else {
+                Ok(())
+            };
+
+            let plaintext = check_keypairs.and_then(|_| {
+                match (task_hpke_keypair, global_hpke_keypair) {
+                    (None, None) => unreachable!("already checked this condition"),
```

Comment:
This is awkward. Why not move the block on lines 1508-1517 into this match arm?

Reply:
The two code paths (`None,None` and the rest) have different errors and behaviors on error. Match arms must return the same type. We _could_ force them to match... but for decryption, we can't directly return `ReportShareError::HpkeDecryptError` because it conceals error context that we need to trace AND we also want to only trace once (not on each failure of trial decryption). I also hoped to avoid copying the tracing calls around, as it's fairly verbose.

We don't have this problem with 1509-1517 because the control flow is simpler, we can just directly return from the function.

I played around with it for a while and this was as concise as I could get, open to other ideas.